pynestml/codegeneration/resources_nest/point_neuron/common/NeuronHeader.jinja2

```diff
@@ -527,6 +527,15 @@ private:
     State_();
   };
 
+{%- if has_delay_variables %}
```

Comment:
Would these fit better in the ``State_`` struct? (Or maybe in ``Buffers_`` — @jougs?)

Reply:
It makes sense to add the state variable's corresponding delayed vector in the `State_` struct. What about the helper variables used to maintain the index and size of the vector?

Reply:
Added all the helper variables for delay-based variables into a new struct named `DelayedVariables_`.